File: obfuscate-examples/ruby/obfuscate.rb
require 'bcrypt'

def hash_sensitive_value (inp)
  puts "Hashing using salt: '#{ENV['BCRYPT_SALT']}'"
  hash = BCrypt::Engine.hash_secret inp, ENV['BCRYPT_SALT']
  puts ".. Full hash: #{hash}"
  BCrypt::Password.new(hash).checksum # Unclear why ruby-bcrypt calls the hash a "checksum"
end

puts "Hashing: '#{ARGV[0]}'"

hashed = hash_sensitive_value ARGV[0]

puts "Final hashed value: '#{hashed}'"

File: obfuscate-examples/java/src/main/java/obfuscate/Obfuscate.java
package obfuscate;

import org.springframework.security.crypto.bcrypt.*;

public class Obfuscate {

  public static String hashSensitiveValue(String inp) {

    String salt = System.getenv("BCRYPT_SALT");
    System.out.println("Hashing using salt: '" + salt + "'");

    String obfuscatedValue = BCrypt.hashpw(inp, salt);
    System.out.println(".. Full hash: " + obfuscatedValue);

    // Strip the algorithm, cost, & salt prefix:
    obfuscatedValue = obfuscatedValue.substring(29);
    return obfuscatedValue;
  }

  public static void main (String[] args) {

    System.out.println("Hashing: '" + args[0] + "'");

    String obfuscatedValue = hashSensitiveValue(args[0]);

    System.out.println("Final hashed value: '" + obfuscatedValue + "'");
  }
}
